### PR TITLE
Url prefix

### PR DESCRIPTION
**Pull request by mmacata**
To make deployment independent of nginx, the openeo-grassgis-driver is - with this PR - able to add the API version prefix itself. As this is an api breaking change when used without nginx, a major release will be necessary afterwards.

### CODE REVIEW

File: tests/test_process_graph_validation.py [hunk 1]
```diff
@@ -68,52 +74,13 @@ def test_6_graph_actiniamodule(self):
...
         """
         response = self.app.post(
+            self.prefix +
             '/validation',
             data=json.dumps(ACTINIA_PROCESS),
             content_type="application/json",
             headers=self.auth)
         self.assertEqual(response.status_code, 200)
 
-    def no_support_test_6_graph_zonal_statistics(self):
```
Comment (metzm) [inline tests/test_process_graph_validation.py:77]: This part has already been deleted from master ?

File: src/openeo_grass_gis_driver/app.py [hunk 2]
```diff
@@ -4,14 +4,18 @@
...
-__author__ = "Sören Gebbert"
-__copyright__ = "Copyright 2018, Sören Gebbert, mundialis"
-__maintainer__ = "Soeren Gebbert"
-__email__ = "soerengebbert@googlemail.com"
+__license__ = "Apache License, Version 2.0"
+__author__ = "Sören Gebbert, Carmen Tawalika"
+__copyright__ = "Copyright 2018-2021, Sören Gebbert, mundialis"
+__maintainer__ = "mundialis"
+
+API_VERSION = "v1.0"
```
Comment (metzm) [inline src/openeo_grass_gis_driver/app.py:12]: Does this mean the driver supports only this specific API version, and API_VERSION would need to be adjusted if `/.well-known/openeo` changes?

**Review by metzm — APPROVED** [review]
I'm not sure about hard-coding API_VERSION independent of well-known, otherwise fine.

File: tests/test_process_graph_validation.py (hunk 1 above)
Comment (mmacata) [reply tests/test_process_graph_validation.py]: I thought I pulled before I started :innocent: I fixed it.

File: src/openeo_grass_gis_driver/app.py (hunk 2 above)
Comment (mmacata) [reply src/openeo_grass_gis_driver/app.py]: I moved the definition to `/.well-known/openeo`. But yes - it means that only one API version is supported which is reflected in the code anyway. If we again need multiple deployments of different versions in the future, we need to work with releases (or branches like in the past).

**Comment by mmacata** [comment]
I moved the definition of API_VERSION to the WellKnown class. Please re-review @metzm

**Review by metzm — APPROVED** [review]
Much better now that the API version is defined only once and used throughout! Please merge.

**Comment by marcjansen** [comment]
Two questions:

* In [these lines in `src/openeo_grass_gis_driver/well_known.py`](https://github.com/Open-EO/openeo-grassgis-driver/pull/85/files#diff-75ded02b57092372d9dd9d3fa1c5b59186314c70f5ef77426d8ae5d66efe8eb2R11-R15) you define one `URL_PREFIX`. Does this mean that one instance of this driver will only ever be able to support one version of the API at the same time? I would have imagined that we may be supporting different versions from one server.
* In [`app.py`](https://github.com/Open-EO/openeo-grassgis-driver/pull/85/files#diff-b2e20df8b10647e0d58a8613479693cdd3b943967bd6dd9cc12f8939afaab63eR18-R20) I read that CORS-support from flask may be shaky, and we therefore use an NGINX before the flask app. Has this been resolved? Dealing in an expected way with CORS is crucial from my understanding of interfaces

**Comment by mmacata** [comment]
> Two questions:
> 
> * In [these lines in `src/openeo_grass_gis_driver/well_known.py`](https://github.com/Open-EO/openeo-grassgis-driver/pull/85/files#diff-75ded02b57092372d9dd9d3fa1c5b59186314c70f5ef77426d8ae5d66efe8eb2R11-R15) you define one `URL_PREFIX`. Does this mean that one instance of this driver will only ever be able to support one version of the API at the same time? I would have imagined that we may be supporting different versions from one server.

Yes that is the case but it has never been different. One state of the backend can only serve one version at a time. When we had multiple versions deployed, we did this by using different branches and let nginx do the reverse proxy. So this change is not a step towards more restriction but towards more transparency in my view. If we ever want to deploy multiple versions again, I would stick to the git branching approach.
> * In [`app.py`](https://github.com/Open-EO/openeo-grassgis-driver/pull/85/files#diff-b2e20df8b10647e0d58a8613479693cdd3b943967bd6dd9cc12f8939afaab63eR18-R20) I read that CORS-support from flask may be shaky, and we therefore use an NGINX before the flask app. Has this been resolved? Dealing in an expected way with CORS is crucial from my understanding of interfaces

The shaky appearance only appeared in combination with nginx and the case that `'Access-Control-Allow-Origin' '*'` was forbidden by the API specification but more restrictive to the requesting client, e.g. `Access-Control-Allow-Origin: http://client.org:8080`. During that time we moved CORS from app.py to nginx because nginx would always add `'Access-Control-Allow-Origin' '*'` unless explicitly hidden, but then also the header from python CORS was hidden. As now `'Access-Control-Allow-Origin' '*'` is fine for a response and we want to get totally rid of nginx, it should be fine but will be tested of course.

**Comment by marcjansen** [comment]
@mmacata sounds all reasonable, thanks. Good to go fdrom my limited point of view.

**Review by metzm — APPROVED** [review]
The explanation of @mmacata for support of one single version makes sense.

**Pull request merged**